pepsico/projections/maproom.py

```diff
@@ -164,6 +164,42 @@ def pick_location(n_clicks, click_lat_lng, region, latitude, longitude):
         return [lat, lng], lat, lng
 
 
+    def local_data(lat, lng, region, model, variable, start_month, end_month):
+        data_ds = xr.Dataset({
+            "histo" : ac.read_data(
+                "historical", model, variable, region, unit_convert=True,
+            ),
+            "picontrol" : ac.read_data(
+                "picontrol", model, variable, region, unit_convert=True,
+            ),
+            "ssp126" : ac.read_data(
+                "ssp126", model, variable, region, unit_convert=True,
+            ),
+            "ssp370" : ac.read_data(
+                "ssp370", model, variable, region, unit_convert=True,
+            ),
+            "ssp585" : ac.read_data(
+                "ssp585", model, variable, region, unit_convert=True,
+            ),
+        })
+        error_msg = None
+        missing_ds = xr.Dataset()
+        try:
+            if any([var is None for var in data_ds.data_vars.values()]):
+                data_ds = missing_ds
+                error_msg="Data missing for this model or variable"
+            else:
+                data_ds = pingrid.sel_snap(data_ds, lat, lng)
+        except KeyError:
```

Comment:
As far as I can see, KeyError can only be raised in line 192. The purpose of the try/except would be clearer if you narrowed its scope to that one line.